packages/jest-util/src/isPromise.ts

```diff
@@ -5,10 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-// capture globalThis.Promise before it may potentially be overwritten
-const Promise = globalThis.Promise;
-
-// see ES2015 spec 25.4.4.5, https://stackoverflow.com/a/38339199
-const isPromise = (candidate: unknown): candidate is Promise<unknown> =>
-  Promise.resolve(candidate) === candidate;
+const isPromise = (candidate: unknown): candidate is PromiseLike<unknown> =>
+  candidate != null &&
+  (typeof candidate === 'object' || typeof candidate === 'function') &&
+  typeof (candidate as any).then === 'function';
```

Comment:
Perhaps? Just to avoid `any`?

```suggestion
  typeof (candidate as PromiseLike<unknown>).then === 'function';
```

Reply:
the cast would lie - I just want to be able to access a prop. I guess `as Record<string, unknown> | (...args: unknown[]) => unknown` or something is more precise looking at the `typeof` checks. But not really, as `typeof new Map === 'object'`. Just saying `any` seems cleanest